src/app/content/components/TableOfContents/styled/index.tsx

```diff
@@ -147,28 +137,64 @@ export const NavOl = styled.ol<{section: ArchiveTree}>`
         }
       }
 
-      & > ${NavItem} > details > ol {
+      & > ${NavItem} > ol {
         margin-left: ${numberWidth + dividerWidth}rem;
       }
+
+      ${/* suppress errors from https://github.com/stylelint/stylelint/issues/3391 */ css`
+        &:not([open]) {
+          display: none;
+
+          a[role='treeitem'] {
+            display: none;
+          }
+
+        }
+      `}
     `;
   }}
 `;
 
-interface DetailsComponentProps {defaultOpen: boolean; open: boolean; }
-class DetailsComponent extends React.Component<DetailsComponentProps, {defaultOpen: boolean}> {
-  constructor(props: DetailsComponentProps) {
-    super(props);
-    this.state = {defaultOpen: props.defaultOpen};
-  }
+interface DetailsComponentProps { open: boolean; }
+class DetailsComponent extends React.Component<DetailsComponentProps> {
   public render() {
-    const {open, defaultOpen: _, ...props} = this.props;
-    const {defaultOpen} = this.state;
+    const { open, ...props } = this.props;
 
-    return <Details {...props} open={open || defaultOpen} />;
+    return <DetailsTree role='treeitem' href="#" {...props} open={open} />;
   }
 }
 
 // tslint:disable-next-line:variable-name
 export const NavDetails = styled(DetailsComponent)`
+  ${labelStyle}
+  display: flex;
   overflow: visible;
+  list-style: none;
+  cursor: pointer;
+  text-decoration: none;
+
+  :focus,
```

Comment:
Lint will complain about "Expected selector .... to come before selector ...."
You can silence that by inserting a comment before this line:
`/* stylelint-disable no-descending-specificity */`

Reply:
There are a lot of lint errors you'll have to fix. Some of them can be done automatically.
First, run `yarn lint` to see where the problems are. Then fix individual files with (for example)
`yarn tslint --fix src/app/content/components/TableOfContents/*.tsx`
That will catch things like wrong quote type and trailing whitespace and missing semicolon.